Pull Request

ZD292 opened the pull request:
# Duke ChatBot 😄 
## [Any fool can write code that a computer can understand](https://www.goodreads.com/author/quotes/25215.Martin_Fowler#:~:text=Martin%20Fowler%20Quotes&text=Any%20fool%20can%20write%20code,code%20that%20humans%20can%20understand.&text=I%27m%20not%20a%20great,good%20programmer%20with%20great%20habits.)
* text-based
* easy to learn
* ~~easy~~ Super easy to use!
## All you need to do is:
1. Download it [here](https://github.com/ZD292/ip)
2. Click it
3. Add you task
4. You are all set!

And its **Free**!

## Features
[x] Managing _Tasks_
[ ] Managing _Deadlines_ (Coming soon)
[ ] _Reminders_ (Coming soon)

If you have some experience in Java, Here is the main method:
```
public static void main(String[] args) {
    new Duke(filePath).run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

ZD292 committed:
L0 greeting implemented

ZD292 committed:
bot can respond now

ZD292 committed:
commit level 2

ZD292 committed:
v3 commit tasks

ZD292 committed:
implemented Events

ZD292 committed:
level 5 handling exceptions

ZD292 committed:
level 6 added a delete function

ZD292 committed:
testing

ZD292 committed:
A-TextUiTesting

ZD292 committed:
testing branch

ZD292 committed:
level 7 branch

ZD292 committed:
Merge branch 'branch-level-7'

ZD292 committed:
fixing merge conflicts

ZD292 committed:
resolve merge conflicts and updated methods

ZD292 committed:
Merge branch 'branch-level-7'

ZD292 committed:
level 8 branch, updated time format

ZD292 committed:
Merge branch 'branch-level-8'

ZD292 committed:
A-MoreOOP and packaging

ZD292 committed:
Merge branch 'A-MoreOOP'

ZD292 committed:
Merge branch 'gradle' into add-gradle-support

ZD292 committed:
gradle Junit and Jar

ZD292 committed:
Merge branch 'add-gradle-support'

ZD292 committed:
Added JavaDocs

ZD292 committed:
Merge branch 'branch-A-JavaDoc'

ZD292 committed:
Adhere to Coding standards

ZD292 committed:
fix merge conflicts

ZD292 committed:
Added FindCommand

ZD292 committed:
fix merge conflits

ZD292 committed:
fixed some edge cases

coderhuang559 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,87 @@
+	private PrintStream out;
+	private static final String COMMENT_LINE_REGEX = "#.";
+
+	private static final String LOGO = " ____        _        \n"
```
Try changing your logo to something else! Maybe try [this](https://www.twitchquotes.com/copypastas/3553). I think it fits your code really well.

coderhuang559 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,87 @@
+			+ "| |_| | |_| |   <  __/\n"
+			+ "|____/ \\__,_|_|\\_\\___|\n";
+
+	private static final String DIVIDER = "____________________________________________________________\n";
```
Great idea to store DIVIDER as constant. Didn't think of that! 😁😊👍

coderhuang559 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,61 @@
+                ui.showLine();
+            }
+        }
+    }
```
Clean `main` and `run`. Good job!

coderhuang559 commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,26 @@
+	 */
+	@Override
+	public void execute(TaskList taskList, Ui u, Storage storage) {
+		String echo = "____________________________________________________________\n" +
```
Would be good to make use of DIVIDER constant throughout project

coderhuang559 reviewed (commented):
Very clean code. Good job!

songgthu commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,36 @@
+	 * @param storage can write tasks to store on the text file.
+	 */
+	@Override
+	public void execute(TaskList taskList, Ui u, Storage storage) {
```
Perhaps you can name the parameter "ui" instead of "u" to be more descriptive?

ZD292 replied:
Thank you for the suggestion 😄

songgthu commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,33 @@
+ * Command to Mark or Unmark task in the taskList.
+ */
+public class MarkCommand extends Command {
+	private final boolean toMark;
```
Maybe you can change the boolean's name to something more intuitive? (e.g: isMarked)

ZD292 replied:
Thank you for the suggestion 😄

songgthu commented on src/main/java/duke/messages/Messages.java:
```diff
@@ -0,0 +1,14 @@
+	public static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format \n%s";
+	public static final String MESSGAE_INIT_FAILED = "Failed to read duke text storage file: \n%s";
+	public static final String MESSGAE_USING_STORAGE = "Using duke text storage file: \n%s";
+	public static final String MESSGAE_INCORRECT_TIME = "Enter time formt either like: \n"
```
There is a small typo "MESSGAE" for the three fields above.

ZD292 replied:
Thank you for pointing that out, I did not see it 😆

songgthu commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,281 @@
+	 */
+	public static String monthValue(String month) throws IllegalArgumentException {
+		switch (month.toLowerCase()) {
+			case "january":
```
Should there be no indentation here for case clauses?

ZD292 replied:
I think my editor auto indented it but I will look into it. Thank you for the suggestion 😄

songgthu commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,281 @@
+		StringBuilder endTime = new StringBuilder();
+
+		switch (commandWord) {
+			case "bye":
```
Same issue as mentioned above.

ZD292 replied:
Thank you for the suggestion 😄

songgthu commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,136 @@
+	 * @param idx Index of task to be marked, starting from 0.
+	 * @throws DukeException If unable to mark task at specified idx.
+	 */
+	public void toMark(Boolean toMark, int idx) throws DukeException {
```
Perhaps you should name the method and argument differently.

ZD292 replied:
Yes that is true, thank you for pointing that out 👍

songgthu reviewed (commented):
Great job for following the coding quality naming conventions. 👍

Goh-Li-Ting commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,26 @@
+		System.out.println(echo);
+		// write to file before leaving the system
+		System.exit(0);
+	}
```
It seems like you have a `showGoodBye` function in `Ui`. Perhaps you can consider using the `showGoodBye` function to implement this `execute` function.

Goh-Li-Ting commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,26 @@
+ */
+public class ByeCommand extends Command {
+	/**
+	 * Executes the bye command which exits system.
```
Perhaps there should be an empty line between description and parameter section? (applies to other Javadoc comments too)

Goh-Li-Ting commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,281 @@
+	 * @return Integer representation of month
+	 * @throws IllegalArgumentException Throws exception should the month not be recognised.
+	 */
+	public static String monthValue(String month) throws IllegalArgumentException {
```
Perhaps it would be better to name this function `getMonthValue` instead so that the method name is a verb?

Goh-Li-Ting commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,281 @@
+	 * @param timeDate Array containing time and date at index 0 and 1.
+	 * @return Formatted String.
+	 */
+	public static String formatDate(String[] timeDate) {
```
Perhaps `timeDate` can be renamed to `timeDates` as this is an array?

Goh-Li-Ting commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,136 @@
+	}
+
+	/**
+	 * Add task to list.
```
Perhaps you should use 'Adds' instead of 'Add'

Goh-Li-Ting commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,136 @@
+	}
+
+	/**
+	 * List all task in taskList in numbered format.
```
Similar to above

Goh-Li-Ting reviewed (commented):
Great work following OOP principles! I added some comments which you can consider!

ZD292 committed:
Add a GUI for easy usage

ZD292 committed:
Code is messy and there exist long methods

Messy code can lead to bugs later on and is harder to understand

Long methods, deeply nested class and large classes are refactored

Refactoring can increase readibility and ease of debugging

Refactored adhering to standard coding practise

ZD292 committed:
Code does not have program checks

Lack of assertions make it harder to debug

Assertions have been added to check length of command words use

Easy to update in the future when extending functionality

ZD292 committed:
User are not able to store tasks for later use

Archive allows users to store and access task

Archive allows for easy task management

Archive includes archive, unarchive, archivelist, archivedelete

Using Archive is intuitive

ZD292 committed:
Ui changes

ZD292 committed:
Handle deadline exception

ZD292 committed:
Reduce nesting

ZD292 committed:
Merge pull request #2 from ZD292/Merging-PRs

ZD292

ZD292 committed:
Add tests for some methods.

Unit test allows the project to expand easily

ZD292 committed:
Update README.md

ZD292 committed:
Update README.md

ZD292 committed:
Update README.md

ZD292 committed:
Update README.md

ZD292 committed:
Update README.md

ZD292 committed:
Update README.md

ZD292 committed:
updated intitialiser

ZD292 committed:
Merge branch 'master' of https://github.com/ZD292/ip

ZD292 committed:
fix ampm bug

ZD292 committed:
edit fxml

ZD292 committed:
add title

ZD292 committed:
fix bug

ZD292 committed:
Update README.md

ZD292 committed:
Update README.md

ZD292 committed:
Update README.md

ZD292 committed:
Update README.md

ZD292 committed:
fix bug

ZD292 committed:
Merge branch 'master' of https://github.com/ZD292/ip

ZD292 committed:
bug fix

ZD292 committed:
Image update

ZD292 committed:
Image update

ZD292 committed:
Delete docs/Ui.png

ZD292 committed:
Add files via upload

ZD292 committed:
add line

ZD292 committed:
Merge pull request #4 from ZD292/branch-Level-10

add line

ZD292 committed:
remove line

ZD292 committed:
Merge pull request #5 from ZD292/branch-A-Assertions

remove line

ZD292 committed:
add line

ZD292 committed:
Merge pull request #6 from ZD292/branch-A-CodeQuality

add line
